test/assembler/encoder.test.ts

```diff
@@ -221,6 +221,107 @@ describe('encode', function () {
       expect(file.content.length).toBe(16)
     }
   })
+
+  it('should encode SPACE instruction with calculated multiplication value', function () {
+    for (const name of ['SPACE', 'FILL', '%']) {
+      const code: ICodeFile = {
+        symbols: {},
+        areas: [
+          {
+            type: AreaType.Data,
+            isReadOnly: true,
+            name: '|.data|',
+            instructions: [
+              {
+                name: name,
+                options: ['8*8'],
+                line: 0
+              }
+            ]
+          }
+        ]
+      }
+      const file = encode(code)
+      expect(Object.keys(file.sections).length).toBe(1)
+      expect(getSection(file, '|.data|').offset).toBe(0)
+      expect(getSection(file, '|.data|').size).toBe(64)
+      expect(file.content.length).toBe(64)
+    }
+  })
+
+  it('should encode SPACE instruction with calculated addition value', function () {
+    for (const name of ['SPACE', 'FILL', '%']) {
+      const code: ICodeFile = {
+        symbols: {},
+        areas: [
+          {
+            type: AreaType.Data,
+            isReadOnly: true,
+            name: '|.data|',
+            instructions: [
+              {
+                name: name,
+                options: ['8   +8'],
+                line: 0
+              }
+            ]
+          }
+        ]
+      }
+      const file = encode(code)
+      expect(Object.keys(file.sections).length).toBe(1)
+      expect(getSection(file, '|.data|').offset).toBe(0)
+      expect(getSection(file, '|.data|').size).toBe(16)
+      expect(file.content.length).toBe(16)
+    }
+  })
+
+  it('should encode SPACE instruction with calculated addition, multiplication and brackets value', function () {
+    for (const name of ['SPACE', 'FILL', '%']) {
```

Comment:
I'm not a huge fan of doing loops within tests. Not that it does not work but in case the test fails it is sometimes hard to find out at which loop position it failed because the test results do not provide this information. So I personally would do three separate tests with a description so you straight away know what functionality exactly failed.

But this is more personal preference and not really a bug per se. So you can also leave it like that.

Reply:
Okay, I will change that.